src/saturn_engine/utils/error_handling.py

```diff
@@ -0,0 +1,188 @@
+from typing import Optional
```

Comment:
nit, but anything that dependes on `saturn_engine.worker` should also be in `saturn_engine.worker`. I would move this to `saturn_engine/worker/error_handler.py`